src/flag_gems/experimental/fft.py

```diff
@@ -0,0 +1,111 @@
+import cmath
+import logging
+
+import numpy as np
+import torch
+import triton
+import triton.language as tl
+
+
+@triton.jit
+def bit_reversal_indices_kernel(
+    output_ptr,
+    n_elements,
+    num_bits,
+    BLOCK_SIZE: tl.constexpr,
+):
+    pid = tl.program_id(axis=0)
+    block_start = pid * BLOCK_SIZE
+    offsets = block_start + tl.arange(0, BLOCK_SIZE)
+    mask = offsets < n_elements
+
+    indices = offsets
+    rev = tl.zeros_like(offsets)
+
+    for j in range(num_bits):
+        rev = tl.where(offsets & (1 << j), rev | (1 << (num_bits - 1 - j)), rev)
+    indices = rev
```

Comment:
We'd consider making num_bits constexpr because this is a fine grained loop and `1 << (num_bits - 1 - j)` be better off with a constant operand.

Reply:
Also, can we remove tl.where predicate by right shifting then left shifting the input bits?
 ```suggestion
    for j in range(num_bits):
        offsets = tl.lshr(offsets, 1)
        rev = rev | tl.shl(offsets & 0x1, num_bits - 1 - j)
    indices = rev
```

Reply:
i don‘t quite understand why those`shifting` can replace `where`, and i didn't find `tl.lshr` in triton 2.2